File: estructura de datos/triangulo.py
import math

class Triangulo:
    def __init__(self, l1=0, l2=0, l3=0):
        self.__l1 = l1
        self.__l2 = l2
        self.__l3 = l3
    def get_l1(self)-> float:
                return self.__l1
            
    def get_l2(self)-> float:
                return self.__l2
            
    def get_l3(self)-> float:
                return self.__l3
            
    def set_l1(self, l1:float)->None:
                self.__l1 = l1
            
    def set_l2(self, l2:float)-> None:
                self.__l2 = l2
            
    def set_l3(self, l3:float)-> None:
                self.__l3 = l3

    def es_triangulo(self)-> bool:
        return self.__l1 + self.__l2 > self.__l3 and self.__l1 + self.__l3 > self.__l2 and self.__l2 + self.__l3 > self.__l1
    
    def area(self):
        s = (self.__l1 + self.__l2 + self.__l3) / 2
        return math.sqrt(s * (s - self.__l1) * (s - self.__l2) * (s - self.__l3))
    
    def perimetro(self):
        return self.__l1 + self.__l2 + self.__l3
    
    def tipo_triangulo(self):
        if self.__l1 == self.__l2 == self.__l3:
            return "Equilátero"
        elif self.__l1 != self.__l2 and self.__l1 != self.__l3 and self.__l2 != self.__l3:
            return "Escaleno"
        else:
            return "Isósceles"
    
    

def main():
    t = Triangulo(l1, l2, l3)
    t.set__l1(float(input("Digite el lado uno del Triángulo: ")))
    l2 = float(input("Digite el lado dos del Triángulo: "))
    l3 = float(input("Digite el lado tres del Triángulo: "))
    if t.es_triangulo():
        print(f"El área del triángulo es: {t.area():.2f}")
        print(f"El perímetro del triángulo es: {t.perimetro():.2f}")
        print(f"El tipo de triángulo es: {t.tipo_triangulo()}")
    else:
        print("No es un triángulo")

if __name__ == "__main__":
    main()

File: estructura de datos/FiguraGeometrica/src/figurageometrica/Triangulo.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package figurageometrica;

/**
 *
 * @author Oscar Perez
 */
public class Triangulo {
    // Declaracion de Atributos
    private double l1,l2,l3;
    //Constructores
    public Triangulo(double l1, double l2, double l3) {
        this.l1 = l1;
        this.l2 = l2;
        this.l3 = l3;
    }
    public Triangulo() {
        this.l1 = 0;
        this.l2 = 0;
        this.l3 = 0;
    }
    //Get
    public double getL1() {
        return l1;
    }
    public double getL2() {
        return l2;
    }
    public double getL3() {
        return l3;
    }
    //Set
    public void setL1(double l1) {
        this.l1 = l1;
    }
    public void setL2(double l2) {
        this.l2 = l2;
    }
    public void setL3(double l3) {
        this.l3 = l3;
    }
    //Operativos
    public boolean esTriangulo(){
        return l1 + l2 > l3 && l1 + l3 > l2 && l2 + l3 > l1;
    }
    public double area(){
        double s,a;
        s = (l1+l2+l3)/2;
        a = Math.sqrt(s*(s-l1)*(s-l2)*(s-l3));
        return a;
    }
    public double perimetro(){
        return l1+l2+l3;
    }
    public String tipoTriangulo(){
        if(this.l1 == this.l2 && this.l1 == this.l3){
            return "Equilatero";
        }
        else{
            if(this.l1 != this.l2 && this.l1 != this.l3 && this.l2 != this.l3){
                return "Escaleno";
            }
            else{
                return "Isoceles";
            }
        }
    }     
}
